### ISSUE TITLE
 Incompatibility with version 4 of MongoDB

### ISSUE THREAD

**Issue by danielwisky** [ff4j/ff4j#465]
Hello guys, how are you?

So, I found an error when I try to persist ff4j in version 4 on mongodb. When I use ff4j-store-mongodb-v3:1.8.10 on my project with mongodb-driver-sync:4.0.5, the application failed to start.

```
***************************
APPLICATION FAILED TO START
***************************
Description:
An attempt was made to call a method that does not exist. The attempt was made from the following location:
    org.ff4j.mongo.store.FeatureStoreMongo.create(FeatureStoreMongo.java:227)
The following method did not exist:
    'void com.mongodb.client.MongoCollection.insertOne(java.lang.Object)'
The method's class, com.mongodb.client.MongoCollection, is available from the following locations:
    jar:file:/home/danielwisky/.m2/repository/org/mongodb/mongodb-driver-sync/4.0.5/mongodb-driver-sync-4.0.5.jar!/com/mongodb/client/MongoCollection.class
The class hierarchy was loaded from the following locations:
    com.mongodb.client.MongoCollection: file:/home/danielwisky/.m2/repository/org/mongodb/mongodb-driver-sync/4.0.5/mongodb-driver-sync-4.0.5.jar
Action:
Correct the classpath of your application so that it contains a single, compatible version of com.mongodb.client.MongoCollection
Disconnected from the target VM, address: '127.0.0.1:33507', transport: 'socket'
Process finished with exit code 1
```

I managed to fix this error compiling the project ff4j-store-mongodb-v3 with version 4.x from mongodb-driver.

```
<dependency>
    <groupId>org.mongodb</groupId>
    <artifactId>mongodb-driver-sync</artifactId>
    <version>4.0.5</version><!--$NO-MVN-MAN-VER$-->
    <scope>provided</scope> <!-- provided to NOT FIX the version -->
</dependency>
```

I'd like your opinion about this fix. I don't know if this change may be done in the same project, ff4j-store-mongodb-v3, or we could create a new one, ff4j-store-mongodb-v4. What dou you guys think about it?

I can help with this changes, if you guys want it.

Thank you, very much.

**Comment by clun** [comment]
I am not surprise this what happened for  v2 and `ff4j-store-mongodb` and why we consider the project v3. 
No complains about v4 except it is not yet compatible with spring boot.

**Comment by danielwisky** [comment]
Hello guys, how are you?
I think ifound another issue. When i complile my project with old version from spring data, the one that uses mongoDb driver v3, using the ff4j-store-mongodb-v3 with mongodb-driver-sync:4.0.5, shows me the following error when i try to persist a feature at my database:

`java.lang.NoSuchMethodError: 'com.mongodb.client.result.InsertOneResult com.mongodb.client.MongoCollection.insertOne(java.lang.Object)'`

When I compile my project with the new version from spring data, the one that uses mongoDb driver v4, using the ff4j-store-mongodb-v3 with mongo-java-driver:3.12.7, shows me the following error when i try to persist a feature at my database:

```
An attempt was made to call a method that does not exist. The attempt was made from the following location:
    org.ff4j.mongo.store.FeatureStoreMongo.create(FeatureStoreMongo.java:227)
The following method did not exist:
    'void com.mongodb.client.MongoCollection.insertOne(java.lang.Object)'
```

If we use both libraries, shows me the following error:


```
The method's class, com.mongodb.client.MongoCollection, is available from the following locations:
    jar:file:/home/danielwisky/.m2/repository/org/mongodb/mongodb-driver-sync/4.0.5/mongodb-driver-sync-4.0.5.jar!/com/mongodb/client/MongoCollection.class
    jar:file:/home/danielwisky/.m2/repository/org/mongodb/mongo-java-driver/3.12.7/mongo-java-driver-3.12.7.jar!/com/mongodb/client/MongoCollection.class
```

I believe we had a compatibility problem. The Big problem here is that if we change to ff4j-store-mongodb-v3(with mongoDb driver v4), we broke the previous versions of spring-data that uses mondoDb driver V3. If we keep the ff4j-store-mongodb-v3(with mongoDb driver v3), we cannot use the spring-data with mongoDb driver V4.

The Spring-Data 2.3.3.RELEASE uses the mondoDb driver V4, I think we can change the ff4j-store-mongodb-v3 to do the same.
What do you guys think?

**Comment by danielwisky** [comment]
I opened a pull-request with a possible solution, another solution would be to create `ff4j-store-mongodb-v4`

**Comment by clun** [comment]
Here I will call help from @drizztguen77 . 
We changed the API 2 weeks ago already for a new version already used by Spring Boot 2.3.x. Moving the interface
Now we change again ? would it be compliant with existing or should we really consider a `store-mongo-v4`

**Comment by drizztguen77** [comment]
@clun I imagine v4 is not backwards compatible with v3 so it might be a good idea to create store-mong-v4. It's a difficult call because if someone is using v3 and we update it to v4 then we force them to upgrade. How is this typically handled with FF4j?

**Comment by drizztguen77** [comment]
Thinking more about it I don't feel we should put the v4 driver into the store-mongodb-v3 package. If we want to support v4 then we should create a new package for v4 or rename the existing store. None of the rest of the stores have a version number on them which means they all just support a single version of a driver. I wonder if we should do the same with mongo. What do you all think?

**Comment by danielwisky** [comment]
I believe that it  makes sense to have a `ff4j-store-mongodb-v4`, but I don’t know if we can keep both versions active simultaneously or if we rename version 3 to 4

**Comment by drizztguen77** [comment]
I just submitted a new PR for this. It will no longer have multiple versions of mongo as that is consistent with all the other feature stores.

https://github.com/ff4j/ff4j/pull/474

**Comment by drizztguen77** [comment]
This has been fixed and merged to master.

**Comment by clun** [comment]
Released in `1.8.11` = thank you @drizztguen77 , again
